src/iron.rs

```diff
@@ -26,7 +26,24 @@ pub struct Iron<H> {
     pub handler: H,
 
     /// Once listening, the local address that this server is bound to.
-    pub addr: Option<SocketAddr>
+    pub addr: Option<SocketAddr>,
+
+    /// Server security mode.
+    pub security: Security
+}
+
+/// Server security mode, either `Insecure` for HTTP or `Secure` for HTTPS.
+pub enum Security {
```

Comment:
Can we discourage matching on this enum exhaustively in the docs? We may later want to add other options, like `Http2`.